woocommerce/woocommerce
# Add Dimensions controls to Product Collection block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces `Width` controls to the `Product Collection` block, enabling users to define fixed or flexible width options.

The implementation adds the width configuration in the block inspector. The controls are placed at the `Product Collection` level and affect the block and its children. So width configuration will affect the `Product Template`, the `Pagination` and the `No Results` blocks.

**Why this change is needed**: This feature enhances flexibility in product display customization, allowing users to better manage the presentation of their product. 

https://github.com/user-attachments/assets/b9678179-49ed-45a1-b00e-3807baf07b28

Closes #47884.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a few products.
2. Navigate to Pages > Add New Page and insert a `Product Collection` block.
3. In the block setup dialog, select `create your own` option.
4. Press the settings button to open the inspector and scroll down to the `Dimensions` options.
5. Under the `Width` options, select `Fixed` and add a value (e.g. 400px).
6. Verify that the `Product Template` block reflects this width immediately.
7. Modify the fixed value and verify that the width changes immediately again.
8. Select the `Fill` option. The `Product Collection` block and its children should fill the available width.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
